plugins/catalog-import/src/components/ImportStepper/defaults.tsx

```diff
@@ -169,74 +170,97 @@ export function defaultGenerateStepper(
-                      error={Boolean(errors.componentName)}
-                      required
-                    />
+                      <Box marginTop={2}>
+                        <Typography variant="h6">
+                          Entity Configuration
+                        </Typography>
+                      </Box>
 
-                    <AutocompleteTextField
-                      name="owner"
-                      control={control}
-                      errors={errors}
-                      options={groups || []}
-                      loading={groupsLoading}
-                      loadingText="Loading groups…"
-                      helperText="Select an owner from the list or enter a reference to a Group or a User"
-                      errorHelperText="required value"
-                      textFieldProps={{
-                        label: 'Entity Owner',
-                        placeholder: 'Group:default/my-group',
-                      }}
-                      rules={{ required: true }}
-                      required
-                    />
-                  </>
-                )}
+                      <TextField
+                        name="componentName"
+                        label="Name of the created component"
+                        placeholder="my-component"
+                        margin="normal"
+                        variant="outlined"
+                        fullWidth
+                        inputRef={register({ required: true })}
+                        error={Boolean(errors.componentName)}
+                        required
+                      />
+
+                      {!watchUseCodeowners && (
+                        <AutocompleteTextField
+                          name="owner"
+                          control={control}
+                          errors={errors}
+                          options={groups || []}
+                          loading={groupsLoading}
+                          loadingText="Loading groups…"
+                          helperText="Select an owner from the list or enter a reference to a Group or a User"
+                          errorHelperText="required value"
+                          textFieldProps={{
+                            label: 'Entity Owner',
+                            placeholder: 'Group:default/my-group',
+                          }}
+                          rules={{ required: true }}
+                          required
+                        />
+                      )}
+
+                      <CheckboxField
+                        name="useCodeowners"
+                        inputRef={register}
+                        label="Use CODEOWNERS File"
+                        helperText="WARNING: This may fail is no CODEOWNERS file is found at the target location."
+                        onChange={(_, value) => {
+                          if (value) {
+                            control.setValue('owner', '');
```

Comment:
Ah I see what you are doing here. You are bypassing the react-hook-form logic. We should definitely fix this later so we don't get splitted logic with unclear side effects.

Reply:
Yeah, I needed some way of resetting the owner. Open to other ideas for this PR as long as they don't require a large amount of refactoring :)